src/cmr/sizing/formats.clj

```diff
@@ -29,20 +37,55 @@
       (get-in [:umm :SizeEstimation :AverageSizeOfGranulesSampled])
       read-number))
 
-(defn- get-avg-compression-rate-ascii
-  "Gets SizeEstimation value for AvgCompressionRateASCII and parses it to a number."
-  [variable]
-  (-> variable
-      (get-in [:umm :SizeEstimation :AvgCompressionRateASCII])
-      read-number))
-
 (defn- get-avg-compression-rate-netcdf4
   "Gets SizeEstimation value for AvgCompressionRateNetCDF4 and parses it to a number."
   [variable]
   (-> variable
       (get-in [:umm :SizeEstimation :AvgCompressionRateNetCDF4])
       read-number))
 
+(defn- get-avg-digit-number
+  "Returns the average of all the numbers in digit-numbers.
+  Return 0 when digit-numbers is empty."
+  [digit-numbers]
+  (if (seq digit-numbers)
+    (/ (reduce + digit-numbers) (count digit-numbers))
+    0))
+
+(defn- get-avg-fill-value-digit-number
+  "Get the average digit number for each value of FillValues in the variable.
+  For example, FillValues containing [{:Value 1} {:Value 100}], returns (1+3)/2=2."
+  [variable]
+  (let [fvs (get-in variable [:umm :FillValues])
+        fvs-digit-numbers (map #(count (str (:Value %))) fvs)]
+    (get-avg-digit-number fvs-digit-numbers)))
+
+(defn- get-neg-to-pos-valid-range-count
+  "Get the negative to postive valid range count."
+  [vrs-min vrs-max]
+  (- (count (filter neg? vrs-min))
+     (+ (count (filter neg? vrs-max))
+        (count (filter zero? vrs-max)))))
```

Comment:
This doesn't count the special case of {:Min 0 :Max 0}. but better than not counting the negative to positive range at all.